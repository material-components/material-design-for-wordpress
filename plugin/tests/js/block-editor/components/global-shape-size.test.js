/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import GlobalShapeSize from '../../../../assets/src/block-editor/components/global-shape-size';

/**
 * Render the component.
 *
 * @param {Object} props - Component props.
 * @return {Function} A functional component.
 */
const setup = props => {
	return render( <GlobalShapeSize { ...props } /> );
};

const baseProps = {
	value: 10,
	onChange: jest.fn(),
	min: 0,
	max: 20,
	blockName: 'material/button',
};

describe( 'GlobalShapeSize', () => {
	beforeEach( () => {
		jest.resetAllMocks();
	} );

	it( 'should match snapshots', () => {
		const props = { ...baseProps };
		expect( setup( props ) ).toMatchSnapshot();

		props.value = 4;
		expect( setup( props ) ).toMatchSnapshot();
	} );
} );
