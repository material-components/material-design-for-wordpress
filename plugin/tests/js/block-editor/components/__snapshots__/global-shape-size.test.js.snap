// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalShapeSize should match snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      hidden="hidden"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="components-base-control components-global-shape-size"
      >
        <div
          class="components-base-control components-range-control"
        >
          <div
            class="components-base-control__field"
          >
            <span
              class="components-range-control__root css-yaebbd-Root exqw8y20"
            >
              <span
                class="components-range-control__wrapper css-okph2p-Wrapper exqw8y21"
                color="#00669b"
              >
                <input
                  aria-hidden="false"
                  class="components-range-control__slider css-15v4a6m-InputRange exqw8y211"
                  id="inspector-range-control-0"
                  max="20"
                  min="0"
                  step="1"
                  tabindex="0"
                  type="range"
                  value="10"
                />
                <span
                  aria-hidden="true"
                  class="css-1lwhlrq-Rail exqw8y24"
                />
                <span
                  aria-hidden="true"
                  class="components-range-control__track css-ogrpie-Track exqw8y25"
                  style="width: 50%;"
                />
                <span
                  class="css-kjao9s-ThumbWrapper exqw8y29"
                  style="left: 50%;"
                >
                  <span
                    aria-hidden="true"
                    class="css-c88ajq-Thumb exqw8y210"
                  />
                </span>
                <span
                  aria-hidden="false"
                  class="components-simple-tooltip components-range-control__tooltip css-1ula2je-Tooltip exqw8y212"
                  role="tooltip"
                  style="left: 50%; z-index: 100;"
                >
                  10
                </span>
              </span>
              <input
                class="components-range-control__number css-1r0c7yf-InputNumber exqw8y213"
                inputmode="decimal"
                max="20"
                min="0"
                step="1"
                type="number"
                value="10"
              />
            </span>
          </div>
        </div>
        <button
          class="components-button global-shape-size-reset is-small"
          type="button"
        >
          Reset
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="components-base-control components-global-shape-size"
    >
      <div
        class="components-base-control components-range-control"
      >
        <div
          class="components-base-control__field"
        >
          <span
            class="components-range-control__root css-yaebbd-Root exqw8y20"
          >
            <span
              class="components-range-control__wrapper css-okph2p-Wrapper exqw8y21"
              color="#00669b"
            >
              <input
                aria-hidden="false"
                class="components-range-control__slider css-15v4a6m-InputRange exqw8y211"
                id="inspector-range-control-0"
                max="20"
                min="0"
                step="1"
                tabindex="0"
                type="range"
                value="10"
              />
              <span
                aria-hidden="true"
                class="css-1lwhlrq-Rail exqw8y24"
              />
              <span
                aria-hidden="true"
                class="components-range-control__track css-ogrpie-Track exqw8y25"
                style="width: 50%;"
              />
              <span
                class="css-kjao9s-ThumbWrapper exqw8y29"
                style="left: 50%;"
              >
                <span
                  aria-hidden="true"
                  class="css-c88ajq-Thumb exqw8y210"
                />
              </span>
              <span
                aria-hidden="false"
                class="components-simple-tooltip components-range-control__tooltip css-1ula2je-Tooltip exqw8y212"
                role="tooltip"
                style="left: 50%; z-index: 100;"
              >
                10
              </span>
            </span>
            <input
              class="components-range-control__number css-1r0c7yf-InputNumber exqw8y213"
              inputmode="decimal"
              max="20"
              min="0"
              step="1"
              type="number"
              value="10"
            />
          </span>
        </div>
      </div>
      <button
        class="components-button global-shape-size-reset is-small"
        type="button"
      >
        Reset
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`GlobalShapeSize should match snapshots 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      hidden="hidden"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="components-base-control components-global-shape-size"
      >
        <div
          class="components-base-control components-range-control"
        >
          <div
            class="components-base-control__field"
          >
            <span
              class="components-range-control__root css-yaebbd-Root exqw8y20"
            >
              <span
                class="components-range-control__wrapper css-okph2p-Wrapper exqw8y21"
                color="#00669b"
              >
                <input
                  aria-hidden="false"
                  class="components-range-control__slider css-15v4a6m-InputRange exqw8y211"
                  id="inspector-range-control-0"
                  max="20"
                  min="0"
                  step="1"
                  tabindex="0"
                  type="range"
                  value="10"
                />
                <span
                  aria-hidden="true"
                  class="css-1lwhlrq-Rail exqw8y24"
                />
                <span
                  aria-hidden="true"
                  class="components-range-control__track css-ogrpie-Track exqw8y25"
                  style="width: 50%;"
                />
                <span
                  class="css-kjao9s-ThumbWrapper exqw8y29"
                  style="left: 50%;"
                >
                  <span
                    aria-hidden="true"
                    class="css-c88ajq-Thumb exqw8y210"
                  />
                </span>
                <span
                  aria-hidden="false"
                  class="components-simple-tooltip components-range-control__tooltip css-1ula2je-Tooltip exqw8y212"
                  role="tooltip"
                  style="left: 50%; z-index: 100;"
                >
                  10
                </span>
              </span>
              <input
                class="components-range-control__number css-1r0c7yf-InputNumber exqw8y213"
                inputmode="decimal"
                max="20"
                min="0"
                step="1"
                type="number"
                value="10"
              />
            </span>
          </div>
        </div>
        <button
          class="components-button global-shape-size-reset is-small"
          type="button"
        >
          Reset
        </button>
      </div>
    </div>
    <div>
      <div
        class="components-base-control components-global-shape-size"
      >
        <div
          class="components-base-control components-range-control"
        >
          <div
            class="components-base-control__field"
          >
            <span
              class="components-range-control__root css-yaebbd-Root exqw8y20"
            >
              <span
                class="components-range-control__wrapper css-okph2p-Wrapper exqw8y21"
                color="#00669b"
              >
                <input
                  aria-hidden="false"
                  class="components-range-control__slider css-15v4a6m-InputRange exqw8y211"
                  id="inspector-range-control-1"
                  max="20"
                  min="0"
                  step="1"
                  tabindex="0"
                  type="range"
                  value="4"
                />
                <span
                  aria-hidden="true"
                  class="css-1lwhlrq-Rail exqw8y24"
                />
                <span
                  aria-hidden="true"
                  class="components-range-control__track css-ogrpie-Track exqw8y25"
                  style="width: 20%;"
                />
                <span
                  class="css-kjao9s-ThumbWrapper exqw8y29"
                  style="left: 20%;"
                >
                  <span
                    aria-hidden="true"
                    class="css-c88ajq-Thumb exqw8y210"
                  />
                </span>
                <span
                  aria-hidden="false"
                  class="components-simple-tooltip components-range-control__tooltip css-1ula2je-Tooltip exqw8y212"
                  role="tooltip"
                  style="left: 20%; z-index: 100;"
                >
                  4
                </span>
              </span>
              <input
                class="components-range-control__number css-1r0c7yf-InputNumber exqw8y213"
                inputmode="decimal"
                max="20"
                min="0"
                step="1"
                type="number"
                value="4"
              />
            </span>
          </div>
        </div>
        <button
          class="components-button global-shape-size-reset is-small"
          type="button"
        >
          Reset
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="components-base-control components-global-shape-size"
    >
      <div
        class="components-base-control components-range-control"
      >
        <div
          class="components-base-control__field"
        >
          <span
            class="components-range-control__root css-yaebbd-Root exqw8y20"
          >
            <span
              class="components-range-control__wrapper css-okph2p-Wrapper exqw8y21"
              color="#00669b"
            >
              <input
                aria-hidden="false"
                class="components-range-control__slider css-15v4a6m-InputRange exqw8y211"
                id="inspector-range-control-1"
                max="20"
                min="0"
                step="1"
                tabindex="0"
                type="range"
                value="4"
              />
              <span
                aria-hidden="true"
                class="css-1lwhlrq-Rail exqw8y24"
              />
              <span
                aria-hidden="true"
                class="components-range-control__track css-ogrpie-Track exqw8y25"
                style="width: 20%;"
              />
              <span
                class="css-kjao9s-ThumbWrapper exqw8y29"
                style="left: 20%;"
              >
                <span
                  aria-hidden="true"
                  class="css-c88ajq-Thumb exqw8y210"
                />
              </span>
              <span
                aria-hidden="false"
                class="components-simple-tooltip components-range-control__tooltip css-1ula2je-Tooltip exqw8y212"
                role="tooltip"
                style="left: 20%; z-index: 100;"
              >
                4
              </span>
            </span>
            <input
              class="components-range-control__number css-1r0c7yf-InputNumber exqw8y213"
              inputmode="decimal"
              max="20"
              min="0"
              step="1"
              type="number"
              value="4"
            />
          </span>
        </div>
      </div>
      <button
        class="components-button global-shape-size-reset is-small"
        type="button"
      >
        Reset
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
